piksi_tools/utils.py

```diff
@@ -38,3 +41,21 @@ def sopen(path, mode):
     '''
     mkdir_p(os.path.dirname(path))
     return open(path, mode)
+
+
+def get_tcp_driver(host, port=None):
+    ''' Factory method helper for opening TCPDriver from host and port
+    '''
+    try:
+        if port is None:
+            host, port = host.split(':')
+        return TCPDriver(host,
+                         int(port),
+                         raise_initial_timeout=True)
+    except ValueError:
+        raise Exception('Invalid format (use ip_address:port): {}'.format(host))
+    except socket.timeout:
+        raise Exception('TCP connection timed out. Check host: {}'.format(host))
+    except Exception as e:
+        import traceback
+        raise Exception('Invalid host and/or port: {0}'.format(traceback.format_exc()))
```

Comment:
Will this be displayed to a user?  Is that really what we want?  Seems like we could at least try to catch some more common errors before displaying a traceback?

Reply:
I believe the current behavior is that the console merely exits, without any dialogue at all. I definitely see your point though, and admittedly, I was keeping this scope relatively focused on avoiding the spinning pinwheel (silent hanging of the connection), rather than refactor the entire feedback cycle.

Reply:
In terms of trying to connect:
1) Invalid host:port format
2) host unreachable
seem to be the most common errors; ValueError and socket.timeout catch them, respectively.

Reply:
I have the feedback method of popping up the "help" dialog when people incorrectly format command line options.   Will this dialog show up in the case of bad / missing host and port?

Reply:
Yes: if command line options are incorrect, the help dialog will pop up before any of this new code gets reached.